admin-console/src/test/java/com/agiletec/apsadmin/admin/TestBaseAdminAction.java

```diff
@@ -1,131 +1,134 @@
-		this.addParameter("newCustomParameter", "parameterValue");
-		this.addParameter("newCustomParameter_newParamMarker", "true");
-		this.addParameter(SystemConstants.CONFIG_PARAM_ERROR_PAGE_CODE, "newErrorPageCode");
-		this.addParameter(SystemConstants.CONFIG_PARAM_HOMEPAGE_PAGE_CODE, "newHomepageCode");
-		result = this.executeAction();
-		assertEquals(Action.SUCCESS, result);
-
-		assertEquals("newHomepageCode", this._configManager.getParam(SystemConstants.CONFIG_PARAM_HOMEPAGE_PAGE_CODE));
-		assertEquals("newErrorPageCode", this._configManager.getParam(SystemConstants.CONFIG_PARAM_ERROR_PAGE_CODE));
-		assertNotNull(this._configManager.getParam("newCustomParameter"));
-		assertEquals("parameterValue", this._configManager.getParam("newCustomParameter"));
-	}
-
-	@Override
-	protected void tearDown() throws Exception {
-		this._configManager.updateConfigItem(SystemConstants.CONFIG_ITEM_PARAMS, this._oldConfigParam);
-		super.tearDown();
-	}
-
-	private void init() {
-		this._configManager = (ConfigInterface) this.getService(SystemConstants.BASE_CONFIG_MANAGER);
-		this._oldConfigParam = this._configManager.getConfigItem(SystemConstants.CONFIG_ITEM_PARAMS);
-	}
-
-	private ConfigInterface _configManager;
-	private String _oldConfigParam;
-
-}
+/*
+ * Copyright 2015-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package com.agiletec.apsadmin.admin;
+
+import java.util.Map;
+
+import com.agiletec.aps.system.SystemConstants;
+import com.agiletec.aps.system.services.baseconfig.ConfigInterface;
+import com.agiletec.apsadmin.ApsAdminBaseTestCase;
+import com.opensymphony.xwork2.Action;
+
+/**
+ * @author E.Santoboni
+ */
+public class TestBaseAdminAction extends ApsAdminBaseTestCase {
+
+    private ConfigInterface configManager;
+    private String oldConfigParam;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        this.init();
+    }
+
+    public void testReloadConfig() throws Throwable {
+        this.setUserOnSession("supervisorCoach");
+        this.initAction("/do/BaseAdmin", "reloadConfig");
+        String result = this.executeAction();
+        assertEquals("userNotAllowed", result);
+
+        this.setUserOnSession("admin");
+        this.initAction("/do/BaseAdmin", "reloadConfig");
+        result = this.executeAction();
+        assertEquals(Action.SUCCESS, result);
+        synchronized (this) {
```

Comment:
Why is this wait needed?

Reply:
As it happens in other tests, with the synchronized block I give the time necessary for the underlying thread to start. Sometimes this test fails when the thread (which sets the expected value) is not started yet